crates/compiler/gen_dev/src/generic64/x86_64.rs

```diff
@@ -935,93 +1241,268 @@ impl CallConv<X86_64GeneralReg, X86_64FloatReg, X86_64Assembler> for X86_64Windo
+        //     mov [rcx + 0x40] r14
+        //     mov [rcx + 0x48] r15
+        //     lea r8 [rsp + 0x08]
+        //     mov [rcx + 0x10] r8
+        //     mov r8 [rsp]
+        //     mov [rcx + 0x50] r8
+        //
+        //     stmxcsr [rcx + 0x58]
+        //     fnstcw word ptr [rcx + 0x5C]
+        //
+        //     mobdxq xmmword ptr [rcx + 0x60], xmm6
+        //     mobdxq xmmword ptr [rcx + 0x70], xmm7
+        //     mobdxq xmmword ptr [rcx + 0x80], xmm8
+        //     mobdxq xmmword ptr [rcx + 0x90], xmm9
+        //     mobdxq xmmword ptr [rcx + 0xa0], xmm10
+        //     mobdxq xmmword ptr [rcx + 0xb0], xmm11
+        //     mobdxq xmmword ptr [rcx + 0xc0], xmm12
+        //     mobdxq xmmword ptr [rcx + 0xd0], xmm13
+        //     mobdxq xmmword ptr [rcx + 0xe0], xmm14
+        //     mobdxq xmmword ptr [rcx + 0xf0], xmm15
+        //
+        //     xor eax, eax
+        //     ret
+
+        let result_pointer = RCX;
+        let env = RDX;
+        debug_assert_eq!(env, Self::GENERAL_PARAM_REGS[1]);
+
+        ASM::mov_mem64_offset32_reg64(buf, env, 0x00, RDX);
+        ASM::mov_mem64_offset32_reg64(buf, env, 0x08, RBX);
+        // NOTE: 0x10 is unused here!
+        ASM::mov_mem64_offset32_reg64(buf, env, 0x18, RBP);
+        ASM::mov_mem64_offset32_reg64(buf, env, 0x20, RSI);
+        ASM::mov_mem64_offset32_reg64(buf, env, 0x28, RDI);
+        ASM::mov_mem64_offset32_reg64(buf, env, 0x30, R12);
+        ASM::mov_mem64_offset32_reg64(buf, env, 0x38, R13);
+        ASM::mov_mem64_offset32_reg64(buf, env, 0x40, R14);
+        ASM::mov_mem64_offset32_reg64(buf, env, 0x48, R15);
+
+        // go one value up (as if setjmp wasn't called)
+        lea_reg64_offset8(buf, R8, RSP, 0x8);
+        ASM::mov_mem64_offset32_reg64(buf, env, 0x10, R8);
+
+        // store the current stack pointer
+        ASM::mov_reg64_mem64_offset32(buf, R8, RSP, 0);
+        ASM::mov_mem64_offset32_reg64(buf, env, 0x50, R8);
+
+        // zero out the fields of the result pointer
+        ASM::mov_reg64_imm64(buf, R8, 0x00);
+        ASM::mov_mem64_offset32_reg64(buf, result_pointer, 0x00, R8);
+        ASM::mov_mem64_offset32_reg64(buf, result_pointer, 0x08, R8);
+
+        // now the windows implementation goes on to store xmm registers and sse2 stuff.
+        // we skip that for now
+
+        // store the result pointer into the env so that longjmp can retrieve it
+        ASM::mov_mem64_offset32_reg64(buf, env, 0x58, result_pointer);
+
+        ASM::ret(buf)
+    }
+
+    fn longjmp(_buf: &mut Vec<'_, u8>) {
+        // do nothing, longjmp is part of roc_panic
+    }
+
+    fn roc_panic(buf: &mut Vec<'_, u8>, relocs: &mut Vec<'_, Relocation>) {
+        use X86_64GeneralReg::*;
+        type ASM = X86_64Assembler;
+
+        // a *const RocStr
+        let roc_str_ptr = R11;
+        ASM::mov_reg64_imm64(buf, roc_str_ptr, 16 + 24); // 24 is width of a rocstr
+        ASM::add_reg64_reg64_reg64(buf, roc_str_ptr, roc_str_ptr, RSP);
```

Comment:
can you shed a light on the constants here? maybe it is related to 
`argument_offset: X86_64WindowsFastcall::SHADOW_SPACE_SIZE as i32 + 16,`?